ARCs/arc-0020.md

```diff
@@ -0,0 +1,870 @@
+---
+arc: 0020
+title: Smart ASA
+description: An ARC for an ASA controlled by an Algorand Smart Contract
+author:
+  - Cosimo Bassi (@cusma)
+  - Adriano Di Luzio (@aldur)
+  - John Jannotti (@jannotti)
+discussions-to: <URL>
+status: Draft
+type: Standards Track
+category (*only required for Standards Track): Interface
+created: 2022-04-27
+requires: ARC-4
+---
+
+## Abstract
+
+A "Smart ASA" is an Algorand Standard Asset (ASA) controlled by a Smart Contract
+that exposes methods to create, configure, transfer, freeze, and destroy the
+asset.
+
+This ARC defines the ABI interface of such Smart Contract, the required
+metadata, and suggests a reference implementation.
+
+## Motivation
+
+The Algorand Standard Asset (ASA) is an excellent building block for on-chain
```

Comment:
Not sure if worth adding, but something I really appreciate about ASAs: As a first class mechanism it also supports a standard, reliable way of representing and querying ownership and history, which is very powerful.